### PR TITLE
WordHighlighter: fix queries against disposed models + blur logic

### LINKED ISSUE #225574: Cannot read properties of null (reading 'modelInfo')
```javascript
TypeError: Cannot read properties of null (reading 'modelInfo')
at z.E in src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts:706:31
at z.restore in src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts:364:8
at V.restoreViewState in src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts:843:26
at u.value in src/vs/workbench/contrib/notebook/browser/view/cellParts/codeCell.ts:317:63
at o.B in src/vs/base/common/event.ts:1230:13
at o.C in src/vs/base/common/event.ts:1241:9
at o.fire in src/vs/base/common/event.ts:1265:9
at _.setValue in src/vs/editor/browser/widget/codeEditor/codeEditorWidget.ts:2028:28
at u.value in src/vs/editor/browser/widget/codeEditor/codeEditorWidget.ts:298:28
at o.B in src/vs/base/common/event.ts:1230:13
at o.fire in src/vs/base/common/event.ts:1261:9
at ee.m in src/vs/editor/browser/widget/codeEditor/codeEditorWidget.ts:2325:19
at u.value in src/vs/editor/browser/widget/codeEditor/codeEditorWidget.ts:2301:9
at o.B in src/vs/base/common/event.ts:1230:13
at o.fire in src/vs/base/common/event.ts:1261:9
at HTMLDivElement.mt in src/vs/base/browser/dom.ts:1349:22
```
[Go to Errors Site](https://errors.code.visualstudio.com/card?ch=eaa41d57266683296de7d118f574d0c2652e1fc4&bH=d213043c-d338-3841-f0be-07a3bff1bb57)

### PR DESCRIPTION
**Pull request by Yoyokrazy**
Fixes #225574

changes:
- no longer do we try to trigger a request against a disposed model, now we store the URI in the query and resolve a textmodel for it
- logic to remove decorations on editor blur
- maintain no flickering if you click within decoration ranges in other editors (when multifile occurrences)

### CODE REVIEW

**Review by rebornix — APPROVED** [review]

File: src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts [hunk 1]
```diff
@@ -675,10 +677,35 @@ class WordHighlighter {
 			// 		1) we have text focus, and a valid query was updated.
 			// 		2) we do not have text focus, and a valid query is cached.
 			// the query will ALWAYS have the correct data for the current highlight request, so it can always be passed to the workerRequest safely
-			if (!WordHighlighter.query || !WordHighlighter.query.modelInfo || WordHighlighter.query.modelInfo.model.isDisposed()) {
+			if (!WordHighlighter.query || !WordHighlighter.query.modelInfo) {
 				return;
 			}
-			this.workerRequest = this.computeWithModel(WordHighlighter.query.modelInfo.model, WordHighlighter.query.modelInfo.selection, WordHighlighter.query.word, otherModelsToHighlight);
+			const queryModelRef = await this.textModelService.createModelReference(WordHighlighter.query.modelInfo.modelURI);
```
Comment (bpasero) [inline src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts:683]: Why is this needed and where is this reference disposed?

File: src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts [hunk 2]
```diff
@@ -675,10 +677,35 @@ class WordHighlighter {
...
+			// this._stopAll(multiFileConfigChange ? this.model.uri : undefined);
+
+			const myRequestId = ++this.workerRequestTokenId;
+			this.workerRequestCompleted = false;
+
+			if (!WordHighlighter.query || !WordHighlighter.query.modelInfo) {
+				return;
+			}
+
+			const queryModelRef = await this.textModelService.createModelReference(WordHighlighter.query.modelInfo.modelURI);
```
Comment (bpasero) [inline src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts:706]: Why is this needed and where is this reference disposed?

File: src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts (hunk 1 above)
Comment (Yoyokrazy) [reply src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts:683]: > Why is this needed

answered in #230635, let me know if you'd like to discuss this problem further.

> where is this reference disposed

to no surprise, given the leak, it wasn't 😞  See #230733 for the fix

File: src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts (hunk 2 above)
Comment (Yoyokrazy) [reply src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts:706]: same as above comment.

**Review by saddam-klg — APPROVED** [review]

**Pull request merged**